account.go

```diff
@@ -53,40 +55,71 @@ func (sc StarkCurve) NewSigner(private, pubX, pubY *big.Int, provider types.Prov
 	- implementation has been tested against OpenZeppelin Account contract as of: https://github.com/OpenZeppelin/cairo-contracts/blob/4116c1ecbed9f821a2aa714c993a35c1682c946e/src/openzeppelin/account/Account.cairo
 	- accepts a multicall
 */
-func (signer *Signer) Execute(ctx context.Context, address string, txs []types.Transaction) (*types.AddTxResponse, error) {
-	nonce, err := signer.provider.AccountNonce(ctx, address)
+func (account *Account) Execute(ctx context.Context, call types.Transaction) (*types.AddTxResponse, error) {
+	calls := []types.Transaction{call}
+
+	nonce, err := account.Provider.AccountNonce(ctx, account.Address)
 	if err != nil {
 		return nil, err
 	}
 
-	maxFee := big.NewInt(0)
-	chainID, err := signer.provider.ChainID(ctx)
-	if err != nil {
-		return nil, err
+	req := types.Transaction{
+		ContractAddress:    account.Address,
+		EntryPointSelector: EXECUTE_SELECTOR,
+		Calldata:           FmtExecuteCalldataStrings(nonce, calls),
 	}
 
-	hash, err := signer.Curve.HashMulticall(address, nonce, maxFee, UTF8StrToBig(chainID), txs)
+	// provide good signature so we can get estimate for ECDSA signing
+	if fee, err := account.EstimateFeeMultiCall(ctx, req, nonce, calls); err == nil {
+		req.MaxFee = fee
+	}
+
+	if r, s, err := account.SignMultiCall(req.MaxFee, nonce, calls); err == nil {
+		req.Signature = []string{r.String(), s.String()}
+	}
```

Comment:
I agree, that would make more sense to write:

```suggestion
        fee, err := account.EstimateFeeMultiCall(ctx, req, nonce, calls); 
	if err != nil {
		return nil, err
	}
	req.MaxFee = fee

	r, s, err := account.SignMultiCall(req.MaxFee, nonce, calls); 
	if err != nil {
		return nil, err
	}
	req.Signature = []string{r.String(), s.String()}
```

or if you want to *continue after an error* without the fee or without the signature for instance, you might want to check for the error. It could be that you have an `expected` error on devnet with the fee but if that is the case, you would have to perform a comparison with a "sentinel error" or with `errors.Is()`.

Reply:
understood. As all '__execute__' functions must pay fees now I'll just return on bad fee estimate and signature